js/app/src/components/HighlightedImage/HighlightedImage.svelte

```diff
@@ -0,0 +1,174 @@
+<script lang="ts">
+	import { createEventDispatcher } from "svelte";
+	import { Block, BlockLabel, Empty } from "@gradio/atoms";
+	import { Image } from "@gradio/icons";
+	import StatusTracker from "../StatusTracker/StatusTracker.svelte";
+	import type { LoadingStatus } from "../StatusTracker/types";
+	import { FileData, normalise_file } from "@gradio/upload";
+	import type { SelectData } from "@gradio/utils";
+	import type { Styles } from "@gradio/utils";
+
+	export let elem_id: string = "";
+	export let elem_classes: Array<string> = [];
+	export let visible: boolean = true;
+	export let value: [FileData, Array<[FileData, string]>] | null;
+	let old_value: [FileData, Array<[FileData, string]>] | null;
+	let _value: [FileData, Array<[FileData, string]>] | null;
+	export let label: string = "Highlighted Image";
+	export let show_label: boolean = true;
+	export let style: Styles = {};
+	export let root: string;
+	export let root_url: string;
+	let active: string | null = null;
+
+	export let loading_status: LoadingStatus;
+
+	const dispatch = createEventDispatcher<{
+		change: undefined;
+		select: SelectData;
+	}>();
+
+	$: {
+		if (value !== old_value) {
+			old_value = value;
+			dispatch("change");
+		}
+		if (value) {
+			_value = [
+				normalise_file(value[0], root, root_url) as FileData,
+				value[1].map(([file, label]) => [
+					normalise_file(file, root, root_url) as FileData,
+					label
+				])
+			];
+		} else {
+			_value = null;
+		}
+	}
+	function handle_mouseover(label: string) {
+		active = label;
+	}
+	function handle_mouseout() {
+		active = null;
+	}
+
+	const FIXED_HEIGHT = 240;
+</script>
+
+<Block
+	{visible}
+	{elem_id}
+	{elem_classes}
+	padding={false}
+	style={{
+		height: style.height || FIXED_HEIGHT,
+		width: style.width
+	}}
+>
+	<StatusTracker {...loading_status} />
+	<BlockLabel {show_label} Icon={Image} label={label || "Image"} />
+
+	<div class="container">
+		{#if _value == null}
+			<Empty size="large" unpadded_box={true}><Image /></Empty>
+		{:else}
+			<div class="image-container">
+				<!-- svelte-ignore a11y-missing-attribute -->
+				<img class="base-image" src={_value ? _value[0].data : null} />
+				{#each _value ? _value[1] : [] as [file, label], i}
+					<!-- svelte-ignore a11y-missing-attribute -->
+					<img
+						class="mask"
+						class:active={active == label}
+						class:inactive={active != label && active != null}
+						src={file.data}
+						style="filter: hue-rotate({_value
+							? Math.round((i * 360) / _value[1].length)
+							: 0}deg)"
+					/>
+				{/each}
+			</div>
+			{#if show_label}
+				<div class="legend">
+					{#each _value ? _value[1] : [] as [file, label], i}
+						<!-- svelte-ignore a11y-click-events-have-key-events -->
+						<div
+							class="legend-item"
+							style="background-color: {_value
```

Comment:
This is very neat. I think users will want to customize their colors (match branding/theme). I think we should add `color_map`? Or maybe the third element in the tuple is the desired color?

Reply:
Done